Summon Guns 2016

Incoming5643 opened the pull request:
Adds a buttload of guns to summon guns to bring it up to speed for 2016. I intentionally skipped any guns that had enough stopping power to OHKO a spessman from full health in one click.

Adds a gun proc, Unlock(), that automatically swaps whatever pin the gun has for a generic electronic one.

:cl: incoming5643
rscadd: The wizard federation has finally updated their assortment of guns for the summon guns event. Look forward to getting shot in the face with a whole new batch of interesting and rarely seen weaponry!
/:cl:

Incoming5643 committed:
Adds a buttload of guns to summon guns to bring it up to speed for 2016. I intentionally skipped any guns that had enough stopping power to OHKO a spessman from full health.

adds a gun proc, Unlock() that automatically swaps whatever pin the gun has for a generic electronic one.

Incoming5643 committed:
Whoops

Bawhoppen commented:
eh if its for wiz I guess

also you should add the smaller version of pulse guns, like pulse pistol

Incoming5643 committed:
whoooooooops

Anonus commented:
Add the xray laser and scatter laser.
Also make the deagle type random from normal deagle, gold-plated deagle, and camo deagle. Deagle.

Incoming5643 commented:
I was firing the scatter weapons and they weren't actually scattering, so I passed.

Shapsy commented:
You have to use them in hand to switch to scatter mode
Scatter lasers are horribly underrated, we need them more

GunHog commented:
I would love to give scatter lasers a new sprite and make them research-able, but I feel the community would defecate on me for trying.

Iamgoofball commented:
do it anyways

Anonus commented on code/modules/projectiles/gun.dm:
```diff
@@ -428,7 +428,9 @@ obj/item/weapon/gun/proc/newshot()
 
-
+/obj/item/weapon/gun/proc/unlock() //used in summon guns and as a convience for admins
+	if(pin)	qdel(pin)
```
don't do this on the same line it's a code standard

Incoming5643 replied:
people have been more anal about "code standards" lately.

I always liked single line if() changes all being on the same line, but whatever.

duncathan replied:
They're standard for a reason
In this case, it's consistency

KazeEspada commented:
Add the 1 shot kill guns, we already have the gyrojet and pulse rifle.

Jordie0608 commented:
Needs a changelog

newfren commented on code/game/gamemodes/wizard/rightandwrong.dm:
```diff
@@ -32,66 +32,90 @@
+					G = new /obj/item/weapon/gun/projectile/revolver/golden(get_turf(H))
+				if("sniper")
+					G = new /obj/item/weapon/gun/projectile/sniper_rifle(get_turf(H))
+				if("medibeam")
```
Incredibly excited to get the medibeam and survivor status.

Iamgoofball replied:
make it a reverse medibeam imo

GunHog replied:
We would have it if Dunc did not hate Science >:-(

Incoming5643 replied:
every roulette needs a booby prize, be thankful I didn't add cap guns instead.

JJRcop replied:
Who says you can't still do that :)

duncathan replied:
@GunHog we can talk when another job gets access to the majority of the cool shit in the game

Science is disproportionately cool; give cool things to other departments too

Jalleo replied:
Or rework R&D to be a multi department thing.

GunHog replied:
It is just really hard to share with other departments. No easy way to send things off to Cargo for redistribution (Flushing it down disposals will just get it spaced). I WANT to give the CMO a Medgun. I WANT to give Sec better guns. I WANT to give the CE NV Mesons and an experimental welder.

But they do not bother to come by Science to pick them up. I cannot be wasting time ferrying things to other people, that means I do not have time to work on the research tree or help out in my department.

KazeEspada replied:
Actually if you wrap something before sending it to cargo, Disposals will reject it from the trash and put it in the mail slot.

Incoming5643 committed:
Adds scatter lasers to the drop list

Changes it so scatter lasers start in scatter mode to drive home the point

CODING STANDARDS

Incoming5643 committed:
Merge branch 'guns_and_guns_and_more_guns' of https://github.com/Incoming5643/-tg-station into guns_and_guns_and_more_guns

# Conflicts:
#	code/game/gamemodes/wizard/rightandwrong.dm

tkdrg commented:
i merged "summon guns 2016" in 2015

Jalleo commented:
This is how its meant to work.
